ledger/catchpointwriter.go

```diff
@@ -138,93 +160,110 @@ func (cw *catchpointWriter) Abort() error {
 	return os.Remove(cw.filePath)
 }
 
+// WriteStep works for a short period of time (determined by stepCtx) to get
+// some more data (accounts/resources/kvpairs) by using readDatabaseStep, and
+// write that data to the open tar file in cw.tar.  The writing is done in
+// asyncWriter, so that it can proceed concurrently with reading the data from
+// the db. asyncWriter only runs long enough to process the data read during a
+// single call to WriteStep, and WriteStep ensures that asyncWriter has finished
+// writing by waiting for it in a defer block, collecting any errors that may
+// have occurred during writing.  Therefore, WriteStep looks like a simple
+// synchronous function to its callers.
 func (cw *catchpointWriter) WriteStep(stepCtx context.Context) (more bool, err error) {
 	// have we timed-out / canceled by that point ?
 	if more, err = hasContextDeadlineExceeded(stepCtx); more || err != nil {
 		return
 	}
 
-	writerRequest := make(chan catchpointFileBalancesChunkV6, 1)
+	writerRequest := make(chan catchpointFileChunkV6, 1)
 	writerResponse := make(chan error, 2)
-	go cw.asyncWriter(writerRequest, writerResponse, cw.balancesChunkNum, cw.numAccountsProcessed)
+	go cw.asyncWriter(writerRequest, writerResponse, cw.chunkNum)
 	defer func() {
+		// For simplicity, all cleanup is done once, here. The writerRequest is
+		// closed, signaling asyncWriter that it can exit, and then
+		// writerResponse is drained, ensuring any problems from asyncWriter are
+		// noted (and that the writing is done).
 		close(writerRequest)
-		// wait for the writerResponse to close.
+	DRAIN:
```

Comment:
I'll change. I think I used all caps from my PERL days!

I'll use lowercase, since that's what I see elsewhere in our codebase, but I will mention that `Drain` seems to be the more common convention elsewhere.

I actually prefer the lowercase convention though, as it seems more inline with the usual Go meaning - this is certainly not a public identifier in any sense.